cerberus/validator.py

```diff
@@ -1038,8 +1058,17 @@ def _validate_keyschema(self, schema, field, value):
     def _validate_readonly(self, readonly, field, value):
         """ {'type': 'boolean'} """
         if readonly:
-            self._error(field, errors.READONLY_FIELD)
-            return True
+            if not self.is_normalized:
+                self._error(field, errors.READONLY_FIELD)
+                return True
+            # If the document was normalized (and therefore already been
+            # checked for readonly fields), we still have to return True
+            # if an error was filed if readonly is a priority rule.
+            if 'readonly' in self.priority_validations:
+                has_error = self.schema_error_tree.fetch_errors_from(
```

Comment:
i have no preference here. you are right that the example case is hardly meaningful, but people come up with uses the issue tracker i'd never thought of regularly.
my way of thinking here is: schemas can be tricky constructs and sometimes rules are evaluated that have not be adressed directly. documents on the other hand are clearer, in the document error tree you will find all errros associated with a field at exactly one node and nowhere else. it seems less error prone to me for the rather weird cases. and as mentioned above, someone will come up with something weird (which is charming).
why would you say the scheme error tree is clearer to understand? genetically they are the same.